extension/wasm_src/src/lib.rs

```
@@ -66,260 +82,18 @@ impl Default for DiagramOptions {
-        let new_node = document.create_element("div")?;
-        new_node.set_attribute("style", "width: 100%;")?;
-
-        let parent_node = match n.parent_node() {
-            Some(node) => node,
-            None => continue,
-        };
-
-        new_node.append_child(&n);
-
-        let (modal_content, modal_container) = create_modal(&document)?;
-        new_node.append_child(modal_container.as_ref());
-
-        // The container which holds the inline-svg on the page
-        let svg_container = Rc::new(document.create_element("div")?);
-        svg_container.set_attribute("class", "railroad_container")?;
-        svg_container.append_child(&document.create_element("svg")?);
-        new_node.append_child(svg_container.as_ref());
-
-        let update_diagram_fn = create_update_fn(
-            Rc::clone(&options),
-            Rc::clone(&svg_container),
-            modal_content,
-            macro_src,
-        )?;
-
-        let icons_container = create_icons(
-            &document,
-            modal_container,
-            Rc::clone(&update_diagram_fn),
-            options,
-        )?;
-
-        svg_container.append_child(&icons_container);
-
-        parent_node.append_child(&new_node);
-        update_diagram_fn()?;
+impl Default for DiagramOptions {
+    fn default() -> Self {
+        Self::new()
     }
-
-    Ok(())
 }
 
 /// Parse the given macro_rules!()-source, returns an SVG and it's preferred width
-fn to_diagram(src: &str, options: &DiagramOptions) -> syn::parse::Result<(i64, String)> {
-    use railroad::RailroadNode;
-
-    let macro_rules = macro_railroad::parser::parse(&src)?;
+#[wasm_bindgen(js_name = toDiagram)]
+pub fn to_diagram(src: &str, options: &DiagramOptions) -> Result<Diagram, JsValue> {
+    log(&format!("{:?}", options));
+    let macro_rules = macro_railroad::parser::parse(src)
+        .map_err(|_| "Macro railroad parse failed.".to_string())?;
```

This will swallow any error from `parse`. Those errors aren't exactly helpful (they come from `syn`), but we should preserve them. Maybe `map_err(|e| format!("macro_railroad parse failed: {}", e)`?!

---

Good idea.